src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -255,33 +268,45 @@ class SymbolTableVisitor : public CommonVisitor<SymbolTableVisitor> {
         bool is_pure = false, is_module = false;
         ASR::accessType s_access = ASR::accessType::Public;
         ASR::deftypeType deftype = ASR::deftypeType::Implementation;
-        char *bindc_name=nullptr;
-        if (sym_name == "main") {
-            // Temporary solution for the main() function
-            std::string return_var_name = "main_return_var";
-            ASR::ttype_t *type = LFortran::ASRUtils::TYPE(ASR::make_Integer_t(al, x.base.base.loc, 8, nullptr, 0));
-            ASR::asr_t *return_var = ASR::make_Variable_t(al, x.base.base.loc,
-                current_scope, s2c(al, return_var_name), ASRUtils::intent_return_var, nullptr, nullptr,
-                ASR::storage_typeType::Default, type,
-                current_procedure_abi_type, ASR::Public, ASR::presenceType::Required,
-                false);
-            LFORTRAN_ASSERT(current_scope->scope.find(return_var_name) == current_scope->scope.end())
-            current_scope->scope[return_var_name]
-                = ASR::down_cast<ASR::symbol_t>(return_var);
-            ASR::asr_t *return_var_ref = ASR::make_Var_t(al, x.base.base.loc,
-                ASR::down_cast<ASR::symbol_t>(return_var));
-
-            tmp = ASR::make_Function_t(
-                al, x.base.base.loc,
-                /* a_symtab */ current_scope,
-                /* a_name */ s2c(al, sym_name),
-                /* a_args */ args.p,
-                /* n_args */ args.size(),
-                /* a_body */ nullptr,
-                /* n_body */ 0,
-                /* a_return_var */ ASRUtils::EXPR(return_var_ref),
-                current_procedure_abi_type,
-                s_access, deftype, bindc_name);
+        char *bindc_name = nullptr;
+
+        if (x.m_returns) {
+            if (AST::is_a<AST::Name_t>(*x.m_returns)) {
+                std::string return_var_name = find_return_var_name(x.n_body, x.m_body);
+                if (return_var_name == "") {
+                    throw SemanticError("Return variable not found",
+                        x.m_returns->base.loc);
+                }
+
+                if(current_scope->scope.find(return_var_name) == current_scope->scope.end()) {
+                    ASR::ttype_t *type = ast_expr_to_asr_type(x.m_returns->base.loc, *x.m_returns);
+                    ASR::asr_t *return_var = ASR::make_Variable_t(al, x.m_returns->base.loc,
+                        current_scope, s2c(al, return_var_name), ASRUtils::intent_return_var, nullptr, nullptr,
+                        ASR::storage_typeType::Default, type,
+                        current_procedure_abi_type, ASR::Public, ASR::presenceType::Required,
+                        false);
+                    current_scope->scope[return_var_name]
+                             = ASR::down_cast<ASR::symbol_t>(return_var);
+                }
```

Comment:
I fixed the double `return` issue. This code also correctly finds the return variable name. Now the issue is if we create the return variable while visiting the `FunctionDef`, there will be a re-declaration error in `AnnAssign`. Any suggestions to tackle this?